rest_framework/serializers.py

```diff
@@ -248,7 +249,36 @@ def errors(self):
         if not hasattr(self, '_errors'):
             msg = 'You must call `.is_valid()` before accessing `.errors`.'
             raise AssertionError(msg)
-        return self._errors
+
+        if isinstance(self._errors, dict):
+            errors = ReturnDict(serializer=self)
+            for key, value in self._errors.items():
+                if isinstance(value, dict):
+                    errors[key] = {}
+                    for key_, value_ in value.items():
+                        message, code = value_[0]
+                        errors[key][key_] = [message]
+                elif isinstance(value, list):
+                    if isinstance(value[0], tuple):
+                        message, code = value[0]
+                    else:
+                        message = value[0]
+                    if isinstance(message, list):
+                        errors[key] = message
+                    else:
+                        errors[key] = [message]
+                elif isinstance(value, tuple):
+                    message, code = value
+                    errors[key] = [message]
+                else:
+                    errors[key] = [value]
+        elif isinstance(self._errors, list):
+            errors = ReturnList(self._errors, serializer=self)
+        else:
+            # This shouldn't ever happen.
+            errors = self._errors
+
+        return errors
```

Comment:
We should find a way to get access to the exception here and just return exc.detail instead of doing all this.

I thought about putting the exception directly in self._errors, I don't know how you feel about that ?